### PR TITLE
Integration of Technologies (App_Analysis) into UI and APIv2

### PR DESCRIPTION
**Pull request by Maffooch**
**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

Previously, managing technologies (renamed from app_analysis) was only accessible through APIv1. Now, they can also be managed through the UI on the product page or APIv2. This was requested by @mtesauro 

Docker-compose dev config also has a change to reduce wait times in auto reloading for processes to be killed off. Instead of the default 60 seconds, processes will now be killed after 3 seconds.

- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted)
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [ ] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [x] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR 

Current accepted labels for PRs:
- Import Scans (for new scanners/importers)
- enhancement
- feature
- bugfix
- maintenance (a.k.a chores)
- dependencies

### CODE REVIEW

**Comment by valentijnscholten** [comment]
I like it. I did notice the api v1 has a breaking (name) change. I am not using it, but we may have to add  a label `breaking-changes` or something.

**Comment by Maffooch** [comment]
Might be better to change it back then since v1 is kind of the legacy api.

**Comment by mtesauro** [comment]
I agree about changing the name of the method for v1 of the API - existing clients will break and, though I'm not thrilled with the name, I'd rather avoid a breaking change.

File: docker/entrypoint-uwsgi-dev.sh [hunk 1]
```diff
@@ -16,4 +16,6 @@ exec uwsgi \
   --enable-threads \
   --processes 2 \
   --threads 2 \
+  --reload-mercy 3 \
+  --worker-reload-mercy 3 \
```
Comment (valentijnscholten) [inline docker/entrypoint-uwsgi-dev.sh:20]: I would set them to 1 to be honest

**Comment by valentijnscholten** [comment]
`django.core.exceptions.FieldDoesNotExist: Risk_Acceptance has no field named 'reporter'` sounds like something went wrong when you merged `dev` into this.

**Comment by Maffooch** [comment]
Just gonna close and start fresh. This is getting gross.

**Pull request closed without merging**